pgwatch2/metrics/settings/10/metric.sql

```diff
@@ -0,0 +1,55 @@
+with qs as (
+  select name, setting from pg_settings
+)
+select /* pgwatch2_generated */
+  (extract(epoch from now()) * 1e9)::int8 as epoch_ns,
+  current_setting('server_version') as server_version,
+  current_setting('server_version_num')::int8 as server_version_num,
+  (regexp_matches(regexp_replace(current_setting('server_version'), '(beta|devel).*', '', 'g'), E'\\d+'))[1]::float8 as major_version,
+  current_setting('block_size')::int as block_size,
+  current_setting('max_connections')::int as max_connections,
+  current_setting('hot_standby') as hot_standby,
+  (select setting from qs where name = 'hot_standby_feedback') as hot_standby_feedback,
+  current_setting('fsync') as fsync,
+  current_setting('full_page_writes') as full_page_writes,
+  current_setting('synchronous_commit') as synchronous_commit,
+  (select setting from qs where name = 'wal_compression') as wal_compression,
```

Comment:
Why cannot we use `select current_setting('wal_compression')` here?

Reply:
Looks ugly I know...but actually has some reasoning behind it...this is a newish parameter and with the current_setting() function you'll get a hard ERROR if the setting is not recognized...

Reply:
Aha, so for old PG versions we're using `SELECT` from CTE and for fresh `current_setting()` directly?